src/host/srvinit.cpp

```diff
@@ -295,6 +333,117 @@ void ConsoleCheckDebug()
+//   errors from the creating the thread for the
+//   standard IO thread loop for the server to process messages
+//   from the driver... or an S_OK success.
+[[nodiscard]] HRESULT ConsoleEstablishHandoff(_In_ HANDLE Server,
+                                              HANDLE driverInputEvent,
+                                              PCONSOLE_API_MSG connectMessage)
+try
+{
+    auto& g = ServiceLocator::LocateGlobals();
+    g.handoffTarget = true;
+
+    IID delegationClsid;
+    if (SUCCEEDED(DelegationConfig::s_GetDefaultConsoleId(delegationClsid)))
+    {
+        g.handoffConsoleClsid = delegationClsid;
+    }
+    if (SUCCEEDED(DelegationConfig::s_GetDefaultTerminalId(delegationClsid)))
+    {
+        g.handoffTerminalClsid = delegationClsid;
+    }
+
+    if (!g.handoffTerminalClsid)
+    {
+        return E_NOT_SET;
+    }
+
+    wil::unique_handle signalPipeTheirSide;
+    wil::unique_handle signalPipeOurSide;
+
+    wil::unique_handle inPipeTheirSide;
+    wil::unique_handle inPipeOurSide;
+
+    wil::unique_handle outPipeTheirSide;
+    wil::unique_handle outPipeOurSide;
+
+    SECURITY_ATTRIBUTES sa;
+    sa.nLength = sizeof(sa);
+    // Mark inheritable for signal handle when creating. It'll have the same value on the other side.
+    sa.bInheritHandle = TRUE;
+    sa.lpSecurityDescriptor = nullptr;
+
+    RETURN_IF_WIN32_BOOL_FALSE(CreatePipe(signalPipeOurSide.addressof(), signalPipeTheirSide.addressof(), nullptr, 0));
+    RETURN_IF_WIN32_BOOL_FALSE(SetHandleInformation(signalPipeTheirSide.get(), HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT));
+
+    RETURN_IF_WIN32_BOOL_FALSE(CreatePipe(inPipeOurSide.addressof(), inPipeTheirSide.addressof(), nullptr, 0));
+    RETURN_IF_WIN32_BOOL_FALSE(SetHandleInformation(inPipeTheirSide.get(), HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT));
+
+    RETURN_IF_WIN32_BOOL_FALSE(CreatePipe(outPipeTheirSide.addressof(), outPipeOurSide.addressof(), nullptr, 0));
+    RETURN_IF_WIN32_BOOL_FALSE(SetHandleInformation(outPipeTheirSide.get(), HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT));
+
+    wil::unique_handle clientProcess{ OpenProcess(PROCESS_QUERY_INFORMATION | SYNCHRONIZE, TRUE, static_cast<DWORD>(connectMessage->Descriptor.Process)) };
+    RETURN_LAST_ERROR_IF_NULL(clientProcess.get());
+
+    ::Microsoft::WRL::ComPtr<ITerminalHandoff> handoff;
+
+    RETURN_IF_FAILED(CoCreateInstance(g.handoffTerminalClsid.value(), nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(&handoff)));
+
+    RETURN_IF_FAILED(handoff->EstablishPtyHandoff(inPipeTheirSide.get(),
+                                                  outPipeTheirSide.get(),
+                                                  signalPipeTheirSide.get(),
+                                                  clientProcess.get()));
+
+    inPipeTheirSide.release();
+    outPipeTheirSide.release();
+    signalPipeTheirSide.release();
+
+    const auto commandLine = fmt::format(L" --headless --signal {:#x}", (int64_t)signalPipeOurSide.release());

```

Comment:
```suggestion
    const auto commandLine = fmt::format(FMT_COMPILE(L" --headless --signal {:#x}"), (int64_t)signalPipeOurSide.release());
```

for the win!

Reply:
I am adequately horrified that we have to use a string to tell our own ConsoleArguments what to do, but it's 2021 and I can't block on that.